torch/csrc/distributed/c10d/ProcessGroup.hpp

```diff
@@ -485,43 +485,16 @@ class TORCH_API ProcessGroup : public torch::CustomClassHolder {
 
   // Agrees on an initial sequence number for the whole group by having rank 0
   // create it and broadcast it to other ranks using the store. Only implemented
-  // for GLOO and NCCL backends currently.
+  // for GLOO, NCCL and UCC backends currently.
   virtual void setSequenceNumberForGroup() {
-    auto backendType = getBackendType();
-    // TODO: HACK for backend name to get sequence number for that backend.
-    if (backendType == ProcessGroup::BackendType::GLOO ||
-        backendType == ProcessGroup::BackendType::NCCL ||
-        backendType == ProcessGroup::BackendType::UCC) {
```

Comment:
I guess the check here is to ensure we only set sequence number for certain backends. Is there any reason why we want to remove this check?

Reply:
@fduwjj  This function is going to call the virtual function with the same name in class Backend here https://github.com/pytorch/pytorch/blob/main/torch/csrc/distributed/c10d/Backend.hpp#L274 
and it  has default implementation with check, so the check is unnecessary.
Further more the check restricts other backends from implementing this function.